universal-application-tool-0.0.1/app/views/LoginForm.java

```diff
@@ -28,11 +28,13 @@
 
   private final BaseHtmlLayout layout;
   private final Config config;
+  private final FakeAdminClient fakeAdminClient;
 
   @Inject
-  public LoginForm(BaseHtmlLayout layout, Config config) {
+  public LoginForm(BaseHtmlLayout layout, Config config, FakeAdminClient fakeAdminClient) {
```

Comment:
Out of curiosity, can you explain why this change is necessary?

Reply:
Prior to this change, `canEnable` was a static method that relied on a hard-coded value that is now provided via start-time configuration. Static methods can't access start-time configuration, so `canEnable` needed to become an instance method, requiring an instance of `fakeAdminClient` be injected here.

As an aside, I generally try to encourage devs working in OO languages to prefer instance methods for public interfaces for this exact reason.

Reply:
Great, thanks so much for this explanation.